gammapy/irf/energy_dispersion.py

```diff
@@ -666,14 +667,14 @@ def from_gauss(cls, e_true, migra, bias, sigma, offset, pdf_threshold=1e-6):
     @classmethod
     def from_table(cls, table):
         """Create from `~astropy.table.Table`."""
-        e_lo = table['ETRUE_LO'].quantity.squeeze()
-        e_hi = table['ETRUE_HI'].quantity.squeeze()
-        o_lo = table['THETA_LO'].quantity.squeeze()
-        o_hi = table['THETA_HI'].quantity.squeeze()
-        m_lo = table['MIGRA_LO'].quantity.squeeze()
-        m_hi = table['MIGRA_HI'].quantity.squeeze()
-
-        matrix = table['MATRIX'].squeeze().transpose()
+        e_lo = table['ETRUE_LO'].quantity[0]
+        e_hi = table['ETRUE_HI'].quantity[0]
+        o_lo = table['THETA_LO'].quantity[0]
+        o_hi = table['THETA_HI'].quantity[0]
+        m_lo = table['MIGRA_LO'].quantity[0]
+        m_hi = table['MIGRA_HI'].quantity[0]
+
+        matrix = table['MATRIX'].quantity[0].transpose() ## TODO Why does this need to be transposed?
```

Comment:
@thomasarmstrong - I think the `EnergyDispersion2D` class should be changed to remove this transpose in the `from_table` and `to_table`, and probably then you also need to update the order of axes as listed in `__init__` and then it should work. That would change the numpy array axis order in memory, but code everywhere in Gammapy should evaluate axes by name and all calling code should still work.

That would still be a temp workaround, I think the final solution would be to parse the axis order from the `CREF ` header key. That would make Gammapy independent of the axis order in the FITS file, we could then read any axis order. For generating new FITS files, a class-level attribute that specifies the "preferred" or "default" axis order would still be needed, i.e. for code that creates IRFs from scratch like you want to in ctapipe.

Previous discussion on IRF axis order is at https://github.com/open-gamma-ray-astro/gamma-astro-data-formats/pull/28 and probably other issues.

My suggestion would be that you try the small change I mentioned and see if `python setup.py test -V` still passes, but leave the larger change involving `CREF` to a future PR. (or directly postpone this to a future PR).